runtime/src/accounts_db.rs

```diff
@@ -176,7 +177,7 @@ impl AccountStorageEntry {
 
         let count = count_and_status.0;
 
-        if status == AccountStorageStatus::StorageFull && count == 0 {
+        if status == AccountStorageStatus::Full && count == 0 {
```

Comment:
Can we just check for `status != AccountStorageStatus::Candidate` here?

Reply:
I don't know.  We'd be continuously calling reset() which re-introduces the race this check fixes?

Reply:
Oh I meant if we added the check `status != AccountStorageStatus::Candidate && count == 0` That way the reset() call shouldn't race with the append() because they are serialized. Either strorage is appending so we won't reset, or reset will happen while we are holding the count and status lock, so we won't append.